tidy3d/plugins/design/method.py

```diff
@@ -23,335 +20,780 @@ class Method(Tidy3dBaseModel, ABC):
             )
-        return val
 
+        # Collapse stores into fn_args and results lists
+        fn_args = [val for sublist in store_parameters for val in sublist]
+        results = [val for sublist in store_fitness for val in sublist]
+
+        return fn_args, results, store_aux, optimizer
+
+
+class AbstractMethodRandom(MethodSample, ABC):
+    """Select parameters with an object with a ``random`` method."""
+
+    num_points: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of Sampling Points",
+        description="The number of points to be generated for sampling.",
+    )
+
+    rng_seed: pd.PositiveInt = pd.Field(
+        default=None,
+        title="Seed",
+        description="Sets the seed used by the optimizers to set constant random number generation.",
+    )
+
+    @abstractmethod
     def get_sampler(self, parameters: Tuple[ParameterType, ...]) -> qmc.QMCEngine:
         """Sampler for this ``Method`` class. If ``None``, sets a default."""
 
-        num_dims_vars = len(parameters)
-        num_dims_sampler = self.sampler.random(1).size
+    def get_run_count(self, parameters: list = None) -> int:
+        return self.num_points
 
-        if num_dims_sampler != num_dims_vars:
-            raise ValueError(
-                f"The sampler {self.sampler} has {num_dims_sampler} dimensions, "
-                f"but the design space has {num_dims_vars} dimensions. These must be equivalent. "
-            )
+    def sample(self, parameters: Tuple[ParameterType, ...], **kwargs) -> Dict[str, Any]:
+        """Defines how the design parameters are sampled on grid."""
+
+        sampler = self.get_sampler(parameters)
+        pts_01 = sampler.random(self.num_points)
+
+        # Convert value from 0-1 to fit within the parameter spans
+        args_by_param = []
+        for i, design_var in enumerate(parameters):
+            pts_i_01 = pts_01[..., i]
+            args_by_param.append(design_var.select_from_01(pts_i_01))
+        args_by_sample = [[row[i] for row in args_by_param] for i in range(len(args_by_param[0]))]
+
+        # Get output list of kwargs for pre_fn
+        keys = [param.name for param in parameters]
+        result = [{keys[j]: row[j] for j in range(len(keys))} for row in args_by_sample]
+
+        return result
+
+
+class MethodMonteCarlo(AbstractMethodRandom):
+    """Select sampling points using Monte Carlo sampling.
+
+    The sampling is done with the Latin Hypercube method from scipy.
+
+    Example
+    -------
+    >>> import tidy3d.plugins.design as tdd
+    >>> method = tdd.MethodMonteCarlo(num_points=20)
+    """
+
+    def get_sampler(self, parameters: Tuple[ParameterType, ...]) -> qmc.QMCEngine:
+        """Sampler for this ``Method`` class."""
 
-        return self.sampler
+        d = len(parameters)
+        return DEFAULT_MONTE_CARLO_SAMPLER_TYPE(d=d, seed=self.rng_seed)
 
 
-MethodType = Union[MethodMonteCarlo, MethodGrid, MethodRandom, MethodRandomCustom]
```

Comment:
I forget, did we want to make `MethodRandom` just subclass from `MethodMonteCarlo` with a warning? or just remove? fine either way.

Reply:
We've removed it now as MonteCarlo is just doing it better.